FEXCore/Source/Interface/Core/JIT/AtomicOps.cpp

```diff
@@ -322,13 +322,26 @@ DEF_OP(AtomicFetchNeg) {
 
   auto MemSrc = GetReg(Op->Addr);
 
-  ARMEmitter::BackwardLabel LoopTop;
-  Bind(&LoopTop);
-  ldaxr(SubEmitSize, TMP2, MemSrc);
-  neg(EmitSize, TMP3, TMP2);
-  stlxr(SubEmitSize, TMP4, TMP3, MemSrc);
-  cbnz(EmitSize, TMP4, &LoopTop);
-  mov(EmitSize, GetReg(Node), TMP2.R());
+  if (CTX->HostFeatures.SupportsAtomics) {
```

Comment:
I'm curious, you have a little unaligned atomic bench. Could you measure the performance between these two paths, aligned and unaligned? Completely uncontended is fine.
I believe the CAS loop will be slower, but I don't have facts to back that up.

Reply:
I assume you want this with userspace atomic handling?

Reply:
Yea, since kernel can't handle the LLSC path anyway.